org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/rewrite/describing/ASTRewritingStringTemplateTest.java

```diff
@@ -442,7 +562,123 @@ public void _test007() throws Exception {
 		buf.append("public class X {\n");
 		buf.append("  void foo(Object o) {\n");
 		buf.append("    String name = \"Jay\";\n");
-		buf.append("    String s = STR.\"\"\"\nHello \\{name}!\"\"\";\n");
+		buf.append("    String s = STR.\"Hello!\";\n");
+		buf.append("  }\n");
+		buf.append("}\n");
+		ICompilationUnit cu= pack1.createCompilationUnit("X.java", buf.toString(), false, null);
+
+		CompilationUnit astRoot= createAST(cu);
+		ASTRewrite rewrite= ASTRewrite.create(astRoot.getAST());
+
+		assertTrue("Parse errors", (astRoot.getFlags() & ASTNode.MALFORMED) == 0);
+		TypeDeclaration type= findTypeDeclaration(astRoot, "X");
+		MethodDeclaration methodDecl= findMethodDeclaration(type, "foo");
+		Block block= methodDecl.getBody();
+		List blockStatements = block.statements();
+		assertEquals("Incorrect number of statements", 2, blockStatements.size());
+		{
+			VariableDeclarationStatement varStmt = (VariableDeclarationStatement) blockStatements.get(1);
+			assertEquals("Incorrect number of fragents", 1, varStmt.fragments().size());
+			VariableDeclarationFragment varFragment = (VariableDeclarationFragment) varStmt.fragments().get(0);
+			StringTemplateExpression templateExp = (StringTemplateExpression) varFragment.getInitializer();
+			rewrite.set(templateExp,  StringTemplateExpression.MULTI_LINE, Boolean.TRUE, null);
+		}
+
+		String preview = evaluateRewrite(cu, rewrite);
+
+		buf= new StringBuilder();
+		buf.append("package test1;\n");
+		buf.append("public class X {\n");
+		buf.append("  void foo(Object o) {\n");
+		buf.append("    String name = \"Jay\";\n");
+		buf.append("    String s = STR.\"\"\"\nHello!\n\"\"\";\n");
+		buf.append("  }\n");
+		buf.append("}\n");
+
+		assertEqualString(preview, buf.toString());
+	}
+
+	@SuppressWarnings({ "rawtypes", "deprecation" })
+	//SINGLE LINE to MULTI LINE with Multiple Components
+	public void test006_c() throws Exception {
```

Comment:
@subyssurendran666 can you please another test with STRing template processor modified with a name having different length than the original template processor